### PR TITLE
doc/dev: add HACKING.rst to Dev Guide

### PR DESCRIPTION
**Pull request by zdover23**
doc/dev: add HACKING.rst to Dev Guide
 
This commit exposes HACKING.rst through the
docs website. Formerly, HACKING.rst was just
a file that existed at /src/pybind/mgr/dashboard/
HACKING.rst. This commit also updates README.rst
to point to the new location of HACKING.rst.

Fixes: https://tracker.ceph.com/issues/47230
Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by LenzGr** [comment]
jenkins render docs

File: src/pybind/mgr/dashboard/README.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ documentation
 for details on how to accomplish this.
 
 Finally, you need to build the dashboard frontend code. See the file
-``HACKING.rst`` in this directory for instructions on setting up the necessary
+:ref:`hacking` for instructions on setting up the necessary
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/README.rst:21]: That reference will likely not work, as the `README.rst` file is not part of the official documentation. I'd suggest to replace this with an URL to the documentation website instead, or a reference to `doc/dev/developer_guide/HACKING.rst` in the git repository.

Comment (LenzGr) [reply src/pybind/mgr/dashboard/README.rst:21]: However, as part of the move into the official Ceph documentation, I'd suggest to use a more intuitive file name other than `HACKING.rst` - how about `dashboard-developer-documentation.rst` instead?

File: src/pybind/mgr/dashboard/README.rst [hunk 2]
```diff
@@ -34,5 +34,5 @@ Working on the Dashboard Code
 -----------------------------
 
 If you're interested in helping with the development of the dashboard, please
-see the file ``HACKING.rst`` for details on how to set up a development
+see :ref:`hacking` for details on how to set up a development
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard/README.rst:37]: Dito here - see above.

File: src/pybind/mgr/dashboard/README.rst (hunk 1 above)
Comment (tchaikov) [reply src/pybind/mgr/dashboard/README.rst:21]: or just `dashboard.rst` as it's located under the `developer_guide` directory already.

**Comment by zdover23** [comment]
jenkins build docs

**Review by jdurgin — APPROVED** [review]

**Comment by LenzGr** [comment]
Thanks for merging! This PR did not remove the original `HACKING.rst` though, so we now have the same documentation on two locations. To avoid confusion, I'll go ahead and replace the original file with a pointer to the new location. I've created [#47396](https://tracker.ceph.com/issues/47396) to track this.

**Pull request merged**